apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java

```diff
@@ -274,9 +280,17 @@ private Coverage transformPartB(Beneficiary beneficiary, Profile profile) {
     if (beneficiary.getBeneEnrollmentReferenceYear().isPresent()) {
       addCoverageDecimalExtension(
           coverage, CcwCodebookVariable.RFRNC_YR, beneficiary.getBeneEnrollmentReferenceYear());
-      // Monthly Medicare-Medicaid dual eligibility codes
-      transformEntitlementDualEligibility(coverage, beneficiary);
 
+      if (profile.equals(Profile.C4DIC)) {
```

Comment:
All of these conditional checks are getting a bit unwieldy. I'm starting to wonder if C4DIC warrants having it's own transform methods. I don't want to add too much scope to what you already have here, but maybe it's something to consider in a subsequent PR.

Reply:
haha . . . I actually love this comment. I did think of refactoring the methods multiple time just to avoid that . . . just seemed way out of the scope the current story

Reply:
Yep, it's tricky because the resources are still _mostly_ similar. So we'd have to figure out the best way to reduce duplication, likely by adding more groups of helper methods.